### PR TITLE
[Feature] support idiom for icon

### PR DESCRIPTION
**Pull request by yyjim**
## Description
Support export icons with `~idiom` suffix.

This PR include change in #38 
## Changes

- Update `ImagePack` type to struct.
- Add `Scale` enum
      - `all`: for all device
      - `individual`: fix scale value
- Update `exportAndroidSVGImages` and `exportAndroidRasterImages`
- Update `XcodeIconsExporter` and `XcodeImagesExporter`
- Add extension functions to create `[FileContents]` for exporting to Xcode.
- Update test cases
- Updated `Examples/Example` assets

### CODE REVIEW

**Comment by subdan** [comment]
This PR breaks color and icon exporting.

1. When I try to export `ic/24/profile~ipad` icon, FigmaExport terminates with the following error:
`Error: Bad asset name «ic_24_profile~ipad»`

2. When I try to export colors to Xcode, FigmaExport terminates with the following error:
`error: Unable to add some file references to Xcode project`

**Comment by subdan** [comment]
Sorry, I've checked wrong branch.

**Comment by subdan** [comment]
Return "@" symbol when exporting images to Xcode project.
image

**Comment by subdan** [comment]
I've run `figma-export images` for iOS example project and the result is wrong.

image
Look at the exclamation marks.

**Comment by yyjim** [comment]
ok, Thanks for review it.
I'll resolve the conflicts and fix the issue soon.

**Comment by yyjim** [comment]
Hi @subdan 
The issues are resolved, please review it when you have time.
Thanks.

**Comment by subdan** [comment]
Thanks. I will make review tomorrow.

File: Sources/XcodeExport/Model/XcodeExportExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+extension AssetPair where AssetType == ImagePack {
+
+    func makeFileContents(to directory: URL, preservesVector: Bool?) throws -> [FileContents] {
+        let name = light.name
+        let dirURL = directory.appendingPathComponent("\(name).imageset")
+
+        let lightPack = light.packForXcode()
+        let darkPack = dark?.packForXcode()
+
+        let properties = { () -> XcodeAssetContents.TemplateProperties? in
```
Comment (subdan) [inline Sources/XcodeExport/Model/XcodeExportExtensions.swift:124]: This closure appears twice in this file. On line 76 and here.

File: Sources/XcodeExport/Model/XcodeExportExtensions.swift [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+            if let preservesVector = preservesVector {
```
Comment (subdan) [inline Sources/XcodeExport/Model/XcodeExportExtensions.swift:125]: Could you try to replace `if else` with `preservesVector?.map { .. }`

File: Sources/XcodeExport/Model/XcodeExportExtensions.swift (hunk 1 above)
Comment (subdan) [reply Sources/XcodeExport/Model/XcodeExportExtensions.swift:124]: You can simplify code from line 124 to 132:
```
let properties = preservesVector.map {
    XcodeAssetContents.TemplateProperties(
        preservesVectorRepresentation: $0 ? true : nil
    )
}
```

File: Sources/XcodeExport/Model/XcodeExportExtensions.swift [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+        }
+
+        if let scaleString = scale.string {
+            urlString.append("@\(scaleString)")
+        }
+
+        return URL(string: urlString)!.appendingPathExtension(format)
+    }
+
+    fileprivate func isValidForXcode(scale: Scale) -> Bool {
```
Comment (subdan) [inline Sources/XcodeExport/Model/XcodeExportExtensions.swift:197]: Why did you choose `fileprivate` instead of `private`?

**Review by subdan — CHANGES_REQUESTED** [review]
Look at my comments.

File: Sources/XcodeExport/Model/XcodeExportExtensions.swift (hunk 3 above)
Comment (yyjim) [reply Sources/XcodeExport/Model/XcodeExportExtensions.swift:197]: The function is called in other extension, so i need to make it `fileprivate`.

File: Sources/XcodeExport/Model/XcodeExportExtensions.swift (hunk 1 above)
Comment (yyjim) [reply Sources/XcodeExport/Model/XcodeExportExtensions.swift:124]: 👍

**Comment by yyjim** [comment]
Hi @subdan 
Please check the last [commit](#7beaca5) for updating the `TemplateProperties`.

**Comment by subdan** [comment]
Thanks for the PR, Jim! I published [a new release](https://github.com/RedMadRobot/figma-export/releases/tag/0.18.0) with your changes.

**Pull request merged**